---
title: "AE_RNAseq"
author: "Amanda Charbonneau"
date: "March 30, 2016"
output: html_document
---


```{r}
require(ggplot2)
require(dplyr)
require(DESeq2)
```
#Import all of the count files from mapping AE reads to a single genome

```{r}
ALLTHEFILES <- dir() #Get file names

File_Num <- length(ALLTHEFILES) 

# Make first dataset into starting dataframe
ALLTHEGENES <- read.csv(ALLTHEFILES[ 1 ], sep = "\t", header = F, 
                        col.names = c("Genes", regmatches(ALLTHEFILES[ 1 ], regexec("[1-6]_2008[0-9]+_[0-9]_[A-Z]+[0-9]", ALLTHEFILES[ 1 ]))))

# Remove X from name of column 2
colnames(ALLTHEGENES) <- c("Genes", substring(colnames(ALLTHEGENES[2]),2))

# In a loop, read in each dataset, then merge it into the starting dataframe

for( X in c( 2:File_Num )){ #start at 2 because first dataset already in

  File_Search <- regexec("[1-6]_2008[0-9]+_[0-9]_[A-Z]+[0-9]", ALLTHEFILES[ X ])
  File_Name <- regmatches(ALLTHEFILES[ X ], File_Search)
  Temp_File <- read.csv(ALLTHEFILES[ X ], sep = "\t", header = F)
  colnames(Temp_File) <- c("Genes", File_Name)
  ALLTHEGENES <- inner_join( ALLTHEGENES, Temp_File )
  
}

meta <- read.csv("../metadata.csv")

```



```{r, echo=FALSE}




```

Note that the `echo = FALSE` parameter was added to the code chunk to prevent printing of the R code that generated the plot.
